### ISSUE TITLE
[RFC] General usage rules and voting rite.

### ISSUE THREAD

**Issue by naroga** [php-cds/php-cds#28]
As of this moment, things are very messy.

People start an idea, and people start voting on said idea right away, without much consideration and even before the actual voting period has started.

I suggest we elaborate a few rules for basic usage. _This has nothing to do with a code of conduct and is not about creating bureaucracy or creating a hierarchy_.

First, I suggest we create two separate issues for a) the discussion and b) the actual vote. The second issue (the vote) would only be opened after the initial discussion period ends (so people will get a chance to see every response before actually reacting). There should be a minimum period of 1-2 weeks of discussion before starting a vote.

Secondly, we should define if people can vote on multiple items.

**Comment by geggleto** [comment]
Ok lets take a look at some of this as it has already been talked about.

1) An RFCs is a request-for-comment, mostly in form of an issue in the php-cds/php-cds repository. Add a title prefixed with [RFC] and start writing about your idea.

2) Once you think you have enough positive feedback from the community (as seen by the Github reactions) you are free to open a discussion thread in the php-cds/php-cds-discussion repository

3)

> Once on a RFC has positive community reaction, you have decided to create a discussion and a concrete implemention, and you think your work is done and ready for prime time, the RFC becomes a PSR-proposal (PHP Standard Recommendation) as soon as you create a issue in the php-cds/php-cds-psr repository.
> 
> Creating an issue in this repository means you want to have a final vote on the RFC asking the community for your proposal to become an official PSR. It's completely up to you how you write your proposal. Any github member is able to express his opinion as a comment or as a vote using :+1: and :-1:.

Some of the items being "voted" on currently are really basic stuff, we are not voting on any actual implementations, but merely how we actually operate the project.

Currently we don't know what Namespace to use or even how long to keep the votes open for!

**Comment by naroga** [comment]
@geggleto, things are so messy right now, that people are considering using `PCS` instead of PHP-CDS, and things are being discussed on the php-cds-discussion repository. 

Let's decide on a few things, like the name and a few ground rules before we can start actually working on things, or this will become another thing nobody takes seriously, with a lot of different "standards" that weren't properly discussed.

**Comment by marcj** [comment]
@naroga I don't think things are messy.
1. We have a vote for "How long is the voting period" https://github.com/php-cds/php-cds/issues/27
2. We have a vote for "What should the namespace be" https://github.com/php-cds/php-cds/issues/4
3. There is already a separation on discussing and voting like you can see in readme. So we have enough time for discussing on real PSRs.
4. Some votes on ground rules like 1) and 2) have been started now, so we don't waste 1 month just to be able to even work. However, everyone is free to discuss to change stuff.

**Comment by marcj** [comment]
Open question

> Secondly, we should define if people can vote on multiple items.

We can not forbid it. It doesn't harm. So I say yes.

**Comment by geggleto** [comment]
Regarding the multiple voting, we are using GitHub's new feature of reactions... I don't think there is a way to stop it currently. Later on down the road I believe that we can have a more formal Polling system that uses GitHub's account... but I think for what we are doing the multiple vote's isn't a bad thing.

**Comment by naroga** [comment]
@geggleto, of course there is a way to stop it. Whoever's moderating the vote can just eliminate votes by people who voted on multiple responses. But I too don't think there is too much harm in letting people vote on multiple choices. Just figured I'd start the discussion in case someone comes up with a good point.

I think this is moving way too fast, without any kind of consideration time. I fear this is going to become that group that tried to standardize EVERYTHING and failed because there were 2352341 votes simultaneously and nobody ever read the entire discussion before actually voting.

There is no need to be hasty here. If we get the 'standards' wrong, this is going to fail miserably. If you want actual community standards, let's push one vote at a time and let people discuss those rules before actually voting.

**Comment by geggleto** [comment]
@naroga The stuff being voted on is purely to setup the stage for actual progress! No actual implementations are being voted on... just a Namespace and how long people think voting should be open for!

**Comment by OddGreg** [comment]
@naroga: Yeah, this is a concern that has passed through my mind a couple of times. I think we are beginning to get our sea legs with this group, and things are progressing in what I presume is the right direction.

**Comment by naroga** [comment]
@geggleto, that's alright, but my opinion is still unchanged.

I just opened my github and there are already 3 new issues being voted on (I don't really care if they are actual standards or just rules), but I feel like it's too much for the first 24hrs. I haven't had the time to completely read everything here yet, and I probably never will be able to follow the speed of these discussions.

Remember, some of the most relevant members of PHP-FIG left because they lacked the time to read every response. The group was "too busy".

And without the support of the framework/library authors, this will never really launch.

Anyway, if you want engaged members, let them breathe.

**Comment by marcj** [comment]
> I just opened my github and there are already 3 new issues being voted on (I don't really care if they are actual standards or just rules),

Btw, there are only 2 active votes for change. Both are fundamental rules, you don't need to read that much and are necessary to be able to work. So we need fast a solution. However, things can chance in the future. So, don't worry, nothing is cast in stone.
